### PR TITLE
Further backport of MTD reconstruction to 10_4_X (continuation of #25541, backport of #25063)

### PR DESCRIPTION
**Pull request by lgray**
As in the title.

This is a backport of the three missing commits to #25541.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test workflow 20434.0

**Comment by lgray** [comment]
@cmsbuild please abort

**Comment by lgray** [comment]
@cmsbuild please test workflow 27434.0

**Comment by fabiocos** [comment]
comparing the results of the merge of #25612 on top of CMSSW_10_5_X and that of #25613 on top of 10_4_X the differences in terms of files are unrelated to MTD and these PRs

**Comment by fabiocos** [comment]
@lgray @civanch the simulation dependency is spurious, wasn't this finally removed in master?

File: SimFastTiming/FastTimingCommon/src/BTLTileDeviceSim.cc [hunk 1]
```diff
@@ -73,7 +73,8 @@ void BTLTileDeviceSim::getHitsResponse(const std::vector<std::tuple<int,uint32_t
       row = btlid.row(topo.nrows());
       col = btlid.column(topo.nrows());	
     }
-
```
Comment (kpedro88) [inline SimFastTiming/FastTimingCommon/src/BTLTileDeviceSim.cc]: @lgray this whitespace change should be removed to be consistent with master

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/25613	f15383c
- the backport of the master version as in https://github.com/cms-sw/cmssw/pull/25063	6e9e477 looks correct (this master version is the same as a squashed "final" version https://github.com/cms-sw/cmssw/pull/25612	f5eb812)
    - part0 of #25063 https://github.com/cms-sw/cmssw/compare/a480cc2041dd20d6ebd2ec67eb050b055f90f728...1663840919f25065abe3bfd2fd2c96a54ce4306e is the same as what's already in 104X as a part of https://github.com/cms-sw/cmssw/pull/25541	3d8d8ba
    - the remaining part of #25063 https://github.com/cms-sw/cmssw/compare/1663840919f25065abe3bfd2fd2c96a54ce4306e...6e9e47711614b34d2576bab37ed78bd90c4b40dc is the same as this PR https://github.com/cms-sw/cmssw/compare/d047c7bd8d74b61b15afec7a6df8f69d20ae8ff7...f15383c0d6d8a20105f21ffc86f2f0dd41e40a2d
- jenkins tests pass and comparisons with the baseline show an expected difference in the ```trackExtenderWithMTD``` collection beta variable, corresponding to the bugfix introduced in this backport
all_oldvsnew_ttbar14tev2023d35wf27434p0c_recotracks_trackextenderwithmtd__reco_obj_beta

**Comment by fabiocos** [comment]
Comparing the files touched by #25063 (or #25612 as they have identical content) and this PR I see two differences:

- the spurious change in the SimFastTiming/FastTimingCommon/src/BTLTileDeviceSim.cc file;
- the difference in DataFormats/TrackerRecHit2D/src/classes_def.xml which looks anyway due to other PRs #25512 https://github.com/cms-sw/cmssw/pull/25512/files#diff-ef4241bc1b10cf2486174cce39090167 

So apart for this blank line (that we may fix afterwards) the PR brings 10_4_X in sync with 10_5_X.

**Comment by fabiocos** [comment]
+operations

**Comment by kpedro88** [comment]
+upgrade

**Comment by kpedro88** [comment]
+1

**Comment by kpedro88** [comment]
merge
simulation signature is spurious, overriding to move ahead with the mtd4 release

**Pull request merged**